### PR TITLE
Kubernetes agent injection on namespaces

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to https://github.com/jaegertracing/jaeger-operator/pull/897 and https://github.com/jaegertracing/jaeger-operator/pull/903/

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: content/docs/next-release/operator.md [hunk 1]
```diff
@@ -549,7 +549,7 @@ Note that the job loads all data for the current day into memory.
 
 ## Auto-injecting Jaeger Agent Sidecars
 
-The operator can inject Jaeger Agent sidecars in `Deployment` workloads, provided that the deployment has the annotation `sidecar.jaegertracing.io/inject` with a suitable value. The values can be either `"true"` (as string), or the Jaeger instance name, as returned by `kubectl get jaegers`. When `"true"` is used, there should be exactly *one* Jaeger instance for the same namespace as the deployment, otherwise, the operator can't figure out automatically which Jaeger instance to use.
+The operator can inject Jaeger Agent sidecars in `Deployment` workloads, provided that the deployment or a namespace has the annotation `sidecar.jaegertracing.io/inject` with a suitable value. The values can be either `"true"` (as string), or the Jaeger instance name, as returned by `kubectl get jaegers`. When `"true"` is used, there should be exactly *one* Jaeger instance for the same namespace as the deployment, otherwise, the operator can't figure out automatically which Jaeger instance to use. A specific Jaeger instance name on a deployment has a higher precedence than `true` applied on a namespace.
```
Comment (objectiser) [inline content/docs/next-release/operator.md]: Wondering if "provided that the deployment or it's namespace has the annotation...." would be more accurate?
"higher precedence than `true` applied on a namespace" change "on a namespace" to "on it's namespace".

Question: I thought the jaeger operator scan's all namespaces not just the namespace in which it resides? So if multiple jaeger operators, then a deployment or namespace would be scanned by multiple operators, so annotation of "true" is ambiguous - so this part of the text may need to be clarified?

Comment (pavolloffay) [reply content/docs/next-release/operator.md]: Yes it should scan all namespaces or namespaces specified in `WATCH_NAMESPACES` env var.

We do not handle a situation when user improperly deploys multiple operators to watch the same set of namespaces.

File: content/docs/next-release/operator.md [hunk 2]
```diff
@@ -549,7 +549,7 @@ Note that the job loads all data for the current day into memory.
...
+The operator can inject Jaeger Agent sidecars in `Deployment` workloads, provided that the deployment or it's namespace has the annotation `sidecar.jaegertracing.io/inject` with a suitable value. The values can be either `"true"` (as string), or the Jaeger instance name, as returned by `kubectl get jaegers`. When `"true"` is used, there should be exactly *one* Jaeger instance for the same namespace as the deployment, otherwise, the operator can't figure out automatically which Jaeger instance to use. A specific Jaeger instance name on a deployment has a higher precedence than `true` applied on it's namespace.
```
Comment (objectiser) [inline content/docs/next-release/operator.md]: Apologies - the two "it's" that I suggested to introduce should be "its" :)

Comment (pavolloffay) [reply content/docs/next-release/operator.md]: that makes more sense, I shouldn't be blindly copying

File: content/docs/next-release/operator.md (hunk 1 above)
Comment (jpkrohling) [reply content/docs/next-release/operator.md]: To clarify: the Jaeger Operator does properly handle CRs in a namespace that is being watched by multiple operators. We store the operator's "owner" in a label as the first thing we do in a reconciliation: 

https://github.com/jaegertracing/jaeger-operator/blob/bce10b1fe5f5cbca176f0d7f49bd287c45d3db60/pkg/controller/jaeger/jaeger_controller.go#L125-L157

That said, this is *not* a supported operation, and users are strongly recommended to have only a single operator watching a given namespace.

**Comment by pavolloffay** [comment]
Can somebody approve this one-liner?

**Pull request merged**